app/validators/form_password_validator.rb

```diff
@@ -2,8 +2,20 @@ module FormPasswordValidator
   extend ActiveSupport::Concern
 
   included do
+    attr_accessor :password
+    attr_reader :user
+
     validates :password,
               presence: true,
               length: Devise.password_length
+
+    validate :valid_password
+  end
+
+  def valid_password
+    return if errors[:password].any?
+    temp_user = User.new(password: password)
+    return if temp_user.valid?
+    errors.add :password, temp_user.errors[:password]
```

Comment:
This is confusing to me as well. Why wouldn't we want to show the user all the errors caused by their entry? If "too short" and "too weak" are unique errors, why not show both if their password is both too short and too weak. If zxcvbn will automatically mark any password that is less than 8 characters long as too weak, then it sounds like we only need one validation, and can get rid of the length validation.